packages/pwa-kit-create-app/scripts/trim-extensions.js

```diff
@@ -0,0 +1,394 @@
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+/* eslint-disable @typescript-eslint/no-var-requires */
+const fs = require('fs')
+const parser = require('@babel/parser')
+const traverse = require('@babel/traverse').default
+const generate = require('@babel/generator').default
+const path = require('path')
+const pluginConfig = require('../assets/plugin-config')
+
+const removeComponentCandidates = [] // List of files that are candidates for removal, as a result of trimming.
+const SEPARATOR = path.sep // Use OS-specific path separator
+const COMPONENT_SCAN_PATHS = [
+    `${SEPARATOR}src${SEPARATOR}components${SEPARATOR}`,
+    `${SEPARATOR}src${SEPARATOR}pages${SEPARATOR}`,
+    `${SEPARATOR}src${SEPARATOR}hooks${SEPARATOR}`
+]
+
+/**
+ * Trim the directory to remove unused components and unused plugins.
+ * @param {*} directory - The directory to trim.
+ * @param {*} generatedPlugins - The plugins that were seleted/unselected by the user.
+ * @returns {void}
+ */
+function trimExtensions(directory, generatedPlugins) {
+    // read plugins from config file
+    const configPlugins = pluginConfig.plugins || {}
+    const plugins = {}
+    Object.keys(configPlugins).forEach((pluginKey) => {
+        plugins[pluginKey] = generatedPlugins?.[pluginKey] || false
+    })
+
+    if (Object.keys(plugins).length === 0) {
+        console.log('No plugins found, skipping trim')
+        return
+    }
+
+    // Helper function to recursively process directory
+    const processDirectory = (dir) => {
+        const files = fs.readdirSync(dir)
+        files.forEach((file) => {
+            const filePath = path.join(dir, file)
+            const stats = fs.statSync(filePath)
+
+            if (stats.isDirectory() && !filePath.includes('node_modules')) {
+                processDirectory(filePath)
+            } else if (file.endsWith('.jsx') || file.endsWith('.tsx')) {
+                processFile(filePath, plugins)
+            }
+        })
+    }
+
+    // Start recursive processing from root directory
+    processDirectory(directory)
+
+    removeUnusedComponents(directory)
+}
+
+function processFile(filePath, plugins) {
+    // Read source file
+    const source = fs.readFileSync(filePath, 'utf-8')
+
+    // Search file content for plugin references
+    const pluginRegex = new RegExp(Object.keys(plugins).join('|'))
+    const hasPluginReferences = pluginRegex.test(source)
+    if (!hasPluginReferences) {
+        // No plugins found in file, return early
+        return false
+    }
+
+    // Parse into AST
+    const ast = parser.parse(source, {
+        sourceType: 'module',
+        plugins: ['jsx', 'typescript']
+    })
+
+    // Track if we made any changes
+    let modified = false
+
+    // Shared helper to evaluate boolean expressions with plugins
+    const evaluateLogicalExpression = (node) => {
```

Comment:
The manual AST is not handling all the scenarios, what about logic operator like `??` or nested ternaries, etc. I think we should use existing babel tools instead.